src/python/evaluation/statistics/common/raw_issue_encoder_decoder.py

```diff
@@ -0,0 +1,82 @@
+    BoolExprLenIssue,
+    CodeIssue,
+    CohesionIssue,
+    CyclomaticComplexityIssue,
+    FuncLenIssue,
+    IssueData,
+    IssueType,
+    LineLenIssue,
+    MaintainabilityLackIssue,
+    Measurable,
+)
+
+MEASURE = 'measure'
+
+
+class RawIssueEncoder(json.JSONEncoder):
+    to_safe_path: bool
+
+    def __init__(self, to_safe_path: bool = True, **kwargs):
+        super().__init__(**kwargs)
+        self.to_safe_path = to_safe_path
+
+    def default(self, obj):
+        if isinstance(obj, BaseIssue):
+            issue_data = {
+                IssueData.ORIGIN_ClASS.value: obj.origin_class,
+                IssueData.ISSUE_TYPE.value: obj.type.value,
+                IssueData.DESCRIPTION.value: obj.description,
+                IssueData.FILE_PATH.value: str(obj.file_path) if self.to_safe_path else "",
+                IssueData.LINE_NUMBER.value: obj.line_no,
+                IssueData.COLUMN_NUMBER.value: obj.column_no,
+                IssueData.INSPECTOR_TYPE.value: obj.inspector_type.value,
+            }
+
+            if isinstance(obj, Measurable):
+                issue_data[MEASURE] = obj.measure()
+
+            return issue_data
+
+        return json.JSONEncoder.default(self, obj)
+
+
+class RawIssueDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kwargs):
+        super().__init__(object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, json_dict):
+        json_dict[IssueData.ISSUE_TYPE.value] = IssueType(json_dict[IssueData.ISSUE_TYPE.value])
+        json_dict[IssueData.INSPECTOR_TYPE.value] = InspectorType(json_dict[IssueData.INSPECTOR_TYPE.value])
+        json_dict[IssueData.FILE_PATH.value] = Path(json_dict[IssueData.FILE_PATH.value])
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.BOOL_EXPR_LEN:
+            json_dict[IssueData.BOOL_EXPR_LEN.value] = json_dict.pop(MEASURE)
+            return BoolExprLenIssue(**json_dict)
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.FUNC_LEN:
+            json_dict[IssueData.FUNCTION_LEN.value] = json_dict.pop(MEASURE)
+            return FuncLenIssue(**json_dict)
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.LINE_LEN:
+            json_dict[IssueData.LINE_LEN.value] = json_dict.pop(MEASURE)
+            return LineLenIssue(**json_dict)
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.CYCLOMATIC_COMPLEXITY:
+            json_dict[IssueData.CYCLOMATIC_COMPLEXITY.value] = json_dict.pop(MEASURE)
+            return CyclomaticComplexityIssue(**json_dict)
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.COHESION:
+            json_dict[IssueData.COHESION_LACK.value] = json_dict.pop(MEASURE)
+            return CohesionIssue(**json_dict)
+
+        if json_dict[IssueData.ISSUE_TYPE.value] == IssueType.MAINTAINABILITY:
+            json_dict[IssueData.MAINTAINABILITY_LACK.value] = json_dict.pop(MEASURE)
+            return MaintainabilityLackIssue(**json_dict)
```

Comment:
Can we rewrite it? Now you have a lot of duplicates

Something like this:
```
       measurable_issues = {
            IssueType.BOOL_EXPR_LEN: BoolExprLenIssue,
            IssueType.FUNC_LEN: FuncLenIssue
        }

        value = json_dict[IssueData.ISSUE_TYPE.value]
        if value in measurable_issues.keys():
            # Get IssueData value by IssueData.ISSUE_TYPE.value
            return measurable_issues[value].__init__(**json_dict)
```

Also measurable_issues dict in this case should be a const not only in this function, maybe in issues.py file